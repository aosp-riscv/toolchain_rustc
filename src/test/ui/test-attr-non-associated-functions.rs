<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
// #[test] attribute is not allowed on associated functions or methods
// reworded error message
// compile-flags:--test

struct A {}

impl A {
    #[test]
    fn new() -> A { //~ ERROR `#[test]` attribute is only allowed on non associated functions
        A {}
    }
}

#[test]
fn test() {
    let _ = A::new();
}

fn main() {}
=======
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
