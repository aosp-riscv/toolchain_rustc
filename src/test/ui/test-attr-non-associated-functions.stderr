<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
error: `#[test]` attribute is only allowed on non associated functions
  --> $DIR/test-attr-non-associated-functions.rs:9:5
   |
LL | /     fn new() -> A {
LL | |         A {}
LL | |     }
   | |_____^

error: aborting due to previous error

=======
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
